### PR TITLE
Pin munki-facts to munkipython and remove all pre 3.6 deprecated code

### PR DESCRIPTION
**Pull request by erikng**
I've reworked all of the munki-facts code to stop supporting deprecated modules

```
DeprecationWarning: The readPlist function is deprecated, use load() instead

DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses

DeprecationWarning: The writePlist function is deprecated, use dump() instead
```

This will mean that munki-facts will only support python 3.6 and higher, but this will also use munki 4.0's built in python, which uses python 3.7. I think given the upcoming changes to macOS, it would be prudent to make this breaking change now and give people a migration path.

### CODE REVIEW

**Comment by erikng** [comment]
Ping on this @gregneagle

**Comment by gregneagle** [comment]
So we need to communicate widely that this change _will_ break any munki-facts you happen to deploy that are not Python-3 compatible.

**Comment by gregneagle** [comment]
Are all the included facts Python 3-ready?

**Comment by erikng** [comment]
Good call. I just tested them and 3 had issues. They are now fixed.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>admin_users</key>
	<array>
		<string>root</string>
		<string>something</string>
	</array>
	<key>backtomymac_configured</key>
	<false/>
	<key>catalina_upgrade_supported</key>
	<false/>
	<key>crashplan_username</key>
	<string>something</string>
	<key>filevault_status</key>
	<string>FileVault is On.</string>
	<key>gatekeeper_status</key>
	<string>assessments enabled</string>
	<key>local_user_dirs</key>
	<array>
		<string>something</string>
	</array>
	<key>mojave_upgrade_supported</key>
	<false/>
	<key>physical_or_virtual</key>
	<string>physical</string>
	<key>sip_status</key>
	<string>System Integrity Protection status: enabled.</string>
</dict>
</plist>
```

The catalina and mojave scripts are out of date, but I don't think it's relevant to fix them for this PR.

**Comment by erikng** [comment]
@gregneagle any thoughts here?

**Comment by gregneagle** [comment]
The shebang line will need to change for Munki 5.1, so let’s make that change now and then merge this after Munki 5.1 ships.

**Comment by erikng** [comment]
Done.

**Comment by natewalck** [comment]
🏓

**Comment by gregneagle** [comment]
As I mentioned before, I'll merge this after the Munki 5.1 release.

**Comment by natewalck** [comment]
Disregard, I forgot. 🤣 I saw an internal PR and pinged this as a result.

**Comment by erikng** [comment]
Let it be known that this internal PR stated this very fact about 5.1.

**Comment by natewalck** [comment]
Let it be known I don't read your comments 🤣

File: munki_facts.py
Comment (natewalck) [inline munki_facts.py:1]: This needs to be `#!/usr/local/munki/munki` now I believe.

**Review by natewalck — CHANGES_REQUESTED** [review]
5.1 Munki path changed back.

File: munki_facts.py
Comment (gregneagle) [reply munki_facts.py:1]: Definitely not that!

(and we're back to `/usr/local/munki/munki-python`...)

Comment (natewalck) [reply munki_facts.py:1]: Make up our minds! 🤣

Comment (natewalck) [reply munki_facts.py:1]: As stated, DISREGARD THIS :D Its fine again.

**Comment by erikng** [comment]
@gregneagle what do I need to do here to get this merged?

**Pull request merged**
